### PR TITLE
Add trusted sources tab.

### PR DESCRIPTION
**Pull request by Smarker**
image

**Corresponding Services PR:** https://github.com/CatalystCode/project-fortis-interfaces/pull/76/files

### CODE REVIEW

File: src/components/Admin/Admin.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ const WATCHLIST_TAB = 1;
 const LOCATIONS_TAB = 2;
```
Comment (erikschlegel) [inline src/components/Admin/Admin.js:21]: We shouldn't be referencing fluxxor in the component, similar to `Dashboard.js`. Please eradicate Fluxxor from the admin module and propagate data via React `props`.

File: src/components/Admin/Admin.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ const WATCHLIST_TAB = 1;
...
 const CUSTOM_EVENTS_TAB = 3;
 const FB_PAGES_TAB = 4;
-const TRUSTED_TWITTER_ACCTS = 5;
+const TRUSTED_SOURCES = 5;
 const TWITTER_ACCOUNTS_TAB = 6;
```
Comment (erikschlegel) [inline src/components/Admin/Admin.js:25]: Please remove any remnants of Facebook and/or Twitter across all admin tabs.

File: src/components/Admin/Admin.js [hunk 3]
```diff
@@ -76,7 +76,7 @@ export const Admin = createReactClass({
                     <Tab>Geofence / Monitored places</Tab>
                     <Tab>Event Import</Tab>
                     <Tab>Facebook pages</Tab>
```
Comment (erikschlegel) [inline src/components/Admin/Admin.js:78]: we dont need facebook pages as trusted sources should support that going forward.

File: src/components/Admin/BlacklistEditor.js [hunk 4]
```diff
@@ -43,7 +43,7 @@ export const BlacklistEditor = createReactClass({
     },
 
     render() {
-      let state = this.getFlux().store("AdminStore").getState();
+      const state = this.getFlux().store("AdminStore").getState();
```
Comment (erikschlegel) [inline src/components/Admin/BlacklistEditor.js:46]: this should be coming through `this.props`

File: src/components/Admin/TrustedSources.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { DataGrid } from './DataGrid';
+import React from 'react';
+import createReactClass from 'create-react-class';
+import Fluxxor from 'fluxxor';
+import constants from '../../actions/constants';
+
+const FluxMixin = Fluxxor.FluxMixin(React);
+const StoreWatchMixin = Fluxxor.StoreWatchMixin("AdminStore");
+
+export const TrustedSources = createReactClass({
```
Comment (erikschlegel) [inline src/components/Admin/TrustedSources.js:10]: remove fluxxor and use ES6 classes

File: src/components/Admin/TrustedSources.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    getAllPipelineKeys() {
+      return constants.DATA_SOURCES.get('all').sourceValues;
+    },
+
+    getStateFromFlux() {
+      return this.getFlux().store("AdminStore").getState();
+    },
+
+    handleSave(rows) {
+      rows.map(row => this.appendRowKey(row));
```
Comment (erikschlegel) [inline src/components/Admin/TrustedSources.js:27]: the list you're transforming isnt being set to any variable?

**Review by erikschlegel — CHANGES_REQUESTED** [review]

File: src/components/Admin/TrustedSources.js (hunk 6 above)
Comment (Smarker) [reply src/components/Admin/TrustedSources.js:27]: Actually this is okay...it still mutates rows and adds the rowKey property I need. If I set `rows.map(row => this.appendRowKey(row));` equal to `rows` I just get the `rowKey` .

File: src/components/Admin/Admin.js (hunk 3 above)
Comment (Smarker) [reply src/components/Admin/Admin.js:78]: ok

File: src/components/Admin/Admin.js (hunk 2 above)
Comment (Smarker) [reply src/components/Admin/Admin.js:25]: ok will do

**Review by erikschlegel — APPROVED** [review]

**Pull request merged**